src/os/bluestore/BlueFS.cc

```diff
@@ -2804,6 +2806,7 @@ void BlueFS::_compact_log_dump_metadata_NF(uint64_t start_seq,
   t->uuid = super.uuid;
 
   std::lock_guard nl(nodes.lock);
+  wal_v2_count = 0;
```

Comment:
We want local copy of `wal_v2_count` here to not influence `encode_transaction()`.
Unless there is some exclusion/mutex around access/modify of `wal_v2_count`, which I do not see.